### PR TITLE
Esmaeel dqm2016 stage2

### PR DESCRIPTION
**Pull request by EsmaeelEskandari**

### CODE REVIEW

**Comment by deguio** [comment]
please test

**Comment by EsmaeelEskandari** [comment]
but I didn't use "L1TSync_Offline.cc" at all

**Comment by davidlange6** [comment]
likely its from your build file changes. What was the motivation for those?

**Comment by EsmaeelEskandari** [comment]
If you mean + 
I commented that because It was complaining about not finding "L1TStage2mGMT" plugin

**Comment by davidlange6** [comment]
maybe you are mixing src and plugins in the same directory?

**Comment by EsmaeelEskandari** [comment]
No, They are in different directories.

**Comment by EsmaeelEskandari** [comment]
I un-commented those lines in BuildFile. Would you please make a test again?

**Comment by deguio** [comment]
did you try to compile locally? did it work?

**Comment by EsmaeelEskandari** [comment]
yes it is compiled completely with or without BuildFile changes.
But when I want to run cmsRun it complains about not finding "L1TStage2mGMT" plugin and when I comment the "export" command in BuildFile it works completely and make plots.

**Comment by EsmaeelEskandari** [comment]
I mean "export" command in BuildFile prevent to produce "L1TStage2mGMT" plugin.

**Comment by deguio** [comment]
please test

File: DQM/Integration/python/clients/l1t2016_dqm_sourceclient-live_cfg-esmaeel.py [hunk 1]
```diff
@@ -0,0 +1,213 @@
+# L1 Trigger DQM sequence (L1T)
```
Comment (deguio) [inline DQM/Integration/python/clients/l1t2016_dqm_sourceclient-live_cfg-esmaeel.py]: this file shouldn't contain your name and should be cleaned from all the commented lines

File: DQM/Integration/python/config/fileinputsource_cfi.py [hunk 2]
```diff
@@ -2,7 +2,9 @@
```
Comment (deguio) [inline DQM/Integration/python/config/fileinputsource_cfi.py]: this one is not supposed to be touched

File: DQM/L1TMonitor/python/L1TStage2_cff-esmaeel.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import FWCore.ParameterSet.Config as cms
```
Comment (deguio) [inline DQM/L1TMonitor/python/L1TStage2_cff-esmaeel.py]: remove your name please

File: DQM/L1TMonitor/src/L1TStage2BMTF.cc [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  
+  // eta ,phi, pt 
+  eta_bmtf = ibooker.book1D("eta_bmtf", "#eta of BMTF", 229, -114.5, 114.5);
+  phi_bmtf = ibooker.book1D("phi_bmtf", "#phi of BMTF", 144, -0.5, 143.5);
+  pt_bmtf = ibooker.book1D("pt_bmtf", "p_{T} of BMTF", 300, 0, 300);
+  bx_bmtf = ibooker.book1D("pt_bmtf", "BX", 2048, -0.5, 2047.5);
+  etaVSphi_bmtf = ibooker.book2D("etaVSphi_bmtf", "#eta VS #phi of BMTF", 229, -114.5, 114.5, 144, -0.5, 143.5);
+  phiVSpt_bmtf = ibooker.book2D("phiVSpt_bmtf", "#phi VS p_{T}of BMTF", 144, -0.5, 143.5, 300, 0, 300);
+  etaVSpt_bmtf = ibooker.book2D("etaVSpt_bmtf", "#eta VS p_{T} of BMTF", 229, -114.5, 114.5,300, 0, 300);
+  etaVSbx_bmtf = ibooker.book2D("etaVSbx_bmtf", "#eta of BMTF VS bx", 229, -114.5, 114.5, 2048, -0.5, 2047.5);
```
Comment (deguio) [inline DQM/L1TMonitor/src/L1TStage2BMTF.cc]: this map and the following ones are very expensive. please review the binning.

File: DQM/Integration/python/config/environment_cfi.py [hunk 5]
```diff
@@ -22,7 +22,7 @@ def loadDQMRunConfigFromFile():
     'userarea': cms.PSet(
         type = cms.untracked.string("userarea"),
         collectorPort = cms.untracked.int32(9190),
-        collectorHost = cms.untracked.string('dqm-c2d07-29.cms'),
+        collectorHost = cms.untracked.string('dqm-c2d07-29.cms'),#lxplus0059.cern.ch'),
```
Comment (deguio) [inline DQM/Integration/python/config/environment_cfi.py]: you are not supposed to touch this

File: DQM/Integration/python/clients/l1t2016_dqm_sourceclient-live_cfg.py [hunk 6]
```diff
@@ -0,0 +1,213 @@
+# L1 Trigger DQM sequence (L1T)
```
Comment (deguio) [inline DQM/Integration/python/clients/l1t2016_dqm_sourceclient-live_cfg.py]: please clean up this config file

**Comment by deguio** [comment]
-1

**Comment by EsmaeelEskandari** [comment]
Would you please make a test?

**Comment by deguio** [comment]
please test

**Comment by EsmaeelEskandari** [comment]
Sorry I took a look at the comparison DQM plots. There are not any DQM plots produced by "L1TStage2mGMT" and "L1TStage2CaloLayer2" ED Analyzers!!

**Comment by EsmaeelEskandari** [comment]
Sorry, what should I do now?

**Comment by deguio** [comment]
the online workflow is not tested here. for that we will need to test at p5.

**Comment by EsmaeelEskandari** [comment]
So is my codes supposed to be put within official CMSSW now?

**Comment by deguio** [comment]
yes. I will review again the code in the next few days and let you know in case something else needs to be changed.

**Comment by deguio** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Comment by EsmaeelEskandari** [comment]
People in mGMT asked me to change the range and bins of histograms which David told me its expensive to have a wide rangw and a lot of bins.
The new proposed ones are:
pt : 0 -- 255  with 0.5 GeV steps  i.e. 510 bins
eta: -2.45 -- 2.45 with 0.011 steps i.e. 445 bins
phi: 0 -- 6.28 with with 0.011 steps i.e. 574 bins
There is the same problem for BMTF and OMTF histograms.
what should I do?

**Comment by deguio** [comment]
please try to contain the binning as much as possible and submit the PR.
I will check if the changes you propose are something we can afford.

cheers,
F.

**Pull request merged**